packages/beacon-node/src/eth1/eth1MergeBlockTracker.ts

```diff
@@ -263,13 +376,73 @@ export class Eth1MergeBlockTracker {
    * TODO: How to prevent doing long recursive search after the merge block has happened?
    */
   private async pollLatestBlock(): Promise<void> {
+    if (this.mergeBlock !== null) {
+      // Merge block already found
+      return;
+    }
+
     const latestBlockRaw = await this.eth1Provider.getBlockByNumber("latest");
     if (!latestBlockRaw) {
       throw Error("getBlockByNumber('latest') returned null");
     }
-
     const latestBlock = toPowBlock(latestBlockRaw);
-    await this.fetchPotentialMergeBlock(latestBlock);
+    this.latestBlock = latestBlock;
+    this.latestBlockTime = quantityToNum(latestBlockRaw.timestamp) * 1000;
+
+    // Set time to merge
+    this.mergeTimeseries.addPoint(
```

Comment:
Adding this TimeSeries code here in the middle of complex logic polutes this class way too much. All this non-critical logic must go somewhere else